src/detect-engine-address-ipv4.c

```diff
@@ -409,854 +409,610 @@ int DetectAddressCutNotIPv4(DetectAddress *a, DetectAddress **b)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "200.200.200.200", &in) < 0);
     a->ip2.addr_data32[0] = in.s_addr;
-    if (inet_pton(AF_INET, "170.170.170.170", &in) < 0)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "170.170.170.170", &in) < 0);
     b->ip.addr_data32[0] = in.s_addr;
-    if (inet_pton(AF_INET, "192.168.1.2", &in) < 0)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "192.168.1.2", &in) < 0);
     b->ip2.addr_data32[0] = in.s_addr;
-    result &= (DetectAddressCmpIPv4(a, b) != ADDRESS_GT);
-
-    DetectAddressFree(a);
-    DetectAddressFree(b);
-    return result;
+    FAIL_IF_NOT(DetectAddressCmpIPv4(a, b) != ADDRESS_GT);
 
- error:
     DetectAddressFree(a);
     DetectAddressFree(b);
-    return 0;
+    PASS;
 }
 
 static int DetectAddressIPv4IsCompleteIPSpace02(void)
 {
     DetectAddress *a = NULL;
     struct in_addr in;
-    int result = 1;
 
-    if ( (a = DetectAddressInit()) == NULL)
-        goto error;
+    a = DetectAddressInit();
+    FAIL_IF_NULL(a);
 
-    if (inet_pton(AF_INET, "0.0.0.0", &in) < 0)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "0.0.0.0", &in) < 0);
     a->ip.addr_data32[0] = in.s_addr;
-    if (inet_pton(AF_INET, "255.255.255.255", &in) < 0)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "255.255.255.255", &in) < 0);
     a->ip2.addr_data32[0] = in.s_addr;
-    result &= (DetectAddressIsCompleteIPSpaceIPv4(a) == 1);
+    FAIL_IF_NOT(DetectAddressIsCompleteIPSpaceIPv4(a) == 1);
 
-    if (inet_pton(AF_INET, "0.0.0.1", &in) < 0)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "0.0.0.1", &in) < 0);
     a->ip.addr_data32[0] = in.s_addr;
-    if (inet_pton(AF_INET, "255.255.255.255", &in) < 0)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "255.255.255.255", &in) < 0);
     a->ip2.addr_data32[0] = in.s_addr;
-    result &= (DetectAddressIsCompleteIPSpaceIPv4(a) == 0);
+    FAIL_IF_NOT(DetectAddressIsCompleteIPSpaceIPv4(a) == 0);
 
     DetectAddressFree(a);
 
-    if ( (a = DetectAddressInit()) == NULL)
-        goto error;
+    a = DetectAddressInit();
+    FAIL_IF_NULL(a);
 
-    if (inet_pton(AF_INET, "0.0.0.0", &in) < 0)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "0.0.0.0", &in) < 0);
     a->ip.addr_data32[0] = in.s_addr;
-    if (inet_pton(AF_INET, "255.255.255.254", &in) < 0)
-        goto error;
+    FAIL_IF(inet_pton(AF_INET, "255.255.255.254", &in) < 0);
     a->ip2.addr_data32[0] = in.s_addr;
-    result &= (DetectAddressIsCompleteIPSpaceIPv4(a) == 0);
+    FAIL_IF_NOT(DetectAddressIsCompleteIPSpaceIPv4(a) == 0);
 
     DetectAddressFree(a);
 
-    return result;
-
- error:
-    if (a != NULL)
-        DetectAddressFree(a);
-    return 0;
+    PASS;
 }
 
 static int DetectAddressIPv4IsCompleteIPSpace03(void)
 {
     DetectAddress *a = NULL;
     DetectAddress *temp = NULL;
     struct in_addr in;
-    int result = 1;
 
-    if ( (a = DetectAddressInit()) == NULL)
-        goto error;
+    a = DetectAddressInit();
```

Comment:
In this and the other tests, we can do like in `DetectAddressIPv4TestAddressCmp01`, and declare the `DetectAdress` variables when we're ready to initialize and check, too, to save up a few more lines (this is a nit, but since there is something else that will require another round, I'm pointing it out :) )